lib/p2p/Pool.ts

```diff
@@ -471,12 +498,14 @@ class Pool extends EventEmitter {
 
     peer.once('open', async () => {
       await this.handleOpen(peer);
+      this.clearOutgoingConnection(peer.nodePubKey);
     });
 
     peer.once('close', () => {
       if (peer.nodePubKey) {
         this.peers.remove(peer.nodePubKey);
       }
+      this.clearOutgoingConnection(peer.nodePubKey);
```

Comment:
I think this line should be inside the `if` statement above, if we don't have a `nodePubKey` value this line won't do anything.